crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -2892,7 +3011,46 @@ impl<
                 self.free_symbol(&tmp);
             }
 
-            x => todo!("getting tag id of union with layout ({:?})", x),
+            UnionLayout::Recursive(_) => {
+                let dst_reg = self.storage_manager.claim_general_reg(&mut self.buf, sym);
+
+                let target_info = self.storage_manager.target_info;
+                if union_layout.stores_tag_id_as_data(target_info) {
+                    let offset = union_layout
+                        .tag_id_offset(self.interner(), target_info)
+                        .unwrap() as i32;
+
+                    let ptr_reg = self
+                        .storage_manager
+                        .load_to_general_reg(&mut self.buf, structure);
+
+                    match union_layout.tag_id_layout() {
+                        Layout::U8 => {
+                            ASM::mov_reg8_mem8_offset32(&mut self.buf, dst_reg, ptr_reg, offset);
+                            ASM::movzx_reg_reg(&mut self.buf, RegisterWidth::W8, dst_reg, dst_reg)
```

Comment:
Just realized this reads a bit unclear by is probably correct. So, IIUC, this is zero filling all 64 bits, despite only theoretically operating on an 8 bit wide register.

Reply:
I believe a pattern match will look at the whole register, so we must make sure that the upper bits are cleared.